public/app/features/plugins/datasource_srv.ts

```diff
@@ -120,6 +120,13 @@ export class DatasourceSrv {
       return 0;
     });
 
+    //at this point grafana and mixed are at the bottom, but depening on the .sort() implementation they can have switched places
```

Comment:
depending on sort in different browsers or depending on phantomsj & node? Is there another sort method we can use? lodash? Would be nice to not have to fix the sort like this.

Reply:
Well, the bug became obvious because of the difference between node and Phantom, but I think the issue is with the specification of `sort()`. Implementations can be different across browsers. It doesn't seem like lodash has a similar sort method, but a rewrite could probably be done to fit one of them.